www/src/hooks/use-active-hash.js

```diff
@@ -0,0 +1,36 @@
+import { useEffect, useState } from "react"
+
+export const useActiveHash = (itemIds, rootMargin = undefined) => {
+  const [activeHash, setActiveHash] = useState(``)
+
+  useEffect(() => {
+    const observer = new IntersectionObserver(
+      entries => {
+        entries.forEach(entry => {
+          if (entry.isIntersecting) {
+            setActiveHash(entry.target.id)
+          }
+        })
+      },
+      { rootMargin: rootMargin || `0% 0% -80% 0%` }
+    )
+
+    itemIds.forEach(id => {
+      observer.observe(document.querySelector(`#${id}`))
```

Comment:
This line will actually break on this page: https://www.gatsbyjs.org/docs/recipes/#1-pages-and-layouts

If you pull it up locally, because its trying to use a querySelector on an invalid CSS selector (like the heading for "#1-pages-and-layouts"). I found [this StackOverflow](https://stackoverflow.com/questions/20306204/using-queryselector-with-ids-that-are-numbers) thread that outlines the problem. If you use `document.querySelector("#\\31-pages-and-layouts")` on that page it will work, I'm not sure what the exact solution should be but we should handle it somehow 🤔

Reply:
Apparently this is OK when `getElementById` is used. Since this is the simplest fix I'm switching to `getElementById`.